### PR TITLE
Changing RMS layer norm to accept DTensors.

### PR DESCRIPTION
**Pull request by kolehma8**
## Summary

Changing RMS layer norm to accept DTensors.

## Details
RMS layer norm parameters are NOT sharded under typical tensor parallelism implementations but the inputs (and gradients) may become sharded (DTensors). This PR resolves the issue by gathering the input tensors and performing the full compute in each device.

Relates Issues:
* https://github.com/linkedin/Liger-Kernel/issues/826
* https://github.com/linkedin/Liger-Kernel/issues/868

## Testing Done

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by shimizust** [comment]
Can you run the example e2e training script just using FSDP2 (since TP won't work until all kernels are tp-compatible) to ensure the perf/loss is correct?

File: src/liger_kernel/ops/rms_norm.py [hunk 1]
```diff
@@ -558,6 +558,13 @@ def forward(ctx, X, W, eps, offset=0.0, casting_mode="llama", in_place=True, row
         X: (B, T, H) or (BxT, H)
         W: (H,)
         """
+        if isinstance(X, torch.distributed.tensor.DTensor):
+            # Input tensor is output of a tensor parallel module and
+            # needs to be gathered to a local tensor to compute
+            # RMSE layer norm on each TP worker.
+            # TODO: support CP.
+            X = X.full_tensor()
```
Comment (shimizust) [inline src/liger_kernel/ops/rms_norm.py:621]: For my understanding, pytorch native TP keeps activations as DTensors and lets subsequent ops decide what to do?

File: src/liger_kernel/utils.py [hunk 2]
```diff
@@ -12,6 +12,33 @@ def is_peft_available():
     return PEFT_AVAILABLE
 
 
+def infer_backend():
```
Comment (shimizust) [inline src/liger_kernel/utils.py]: Suggest renaming to `infer_comm_backend`

File: src/liger_kernel/ops/rms_norm.py (hunk 1 above)
Comment (kolehma8) [reply src/liger_kernel/ops/rms_norm.py:621]: That is my understanding as well.

File: src/liger_kernel/utils.py (hunk 2 above)
Comment (kolehma8) [reply src/liger_kernel/utils.py]: Changed as suggested.

**Comment by kolehma8** [comment]
> Can you run the example e2e training script just using FSDP2 (since TP won't work until all kernels are tp-compatible) to ensure the perf/loss is correct?

I did not see any benchmarks using FSDP in the scripts folder. I think we can say to a very high degree of confidence that this PR will not impact FSDP since if it did, the op would have crashed earlier (e.g. if any inputs are DTensors). Vice versa, this PR only impacts input DTensors and behavior for regular tensors remains unchanged.
